src/main/java/seedu/wife/logic/commands/foodcommands/ListByTagCommand.java

```diff
@@ -0,0 +1,63 @@
+package seedu.wife.logic.commands.foodcommands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.wife.commons.util.TagUtil;
+import seedu.wife.logic.commands.Command;
+import seedu.wife.logic.commands.CommandResult;
+import seedu.wife.model.Model;
+import seedu.wife.model.tag.Tag;
+
+/**
+ * Lists all persons in the address book to the user.
+ */
+public class ListByTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "listbytag";
+    public static final String MESSAGE_SUCCESS = "Listed all food with the following tags:";
+    public static final String MESSAGE_NO_FOOD_WITH_FOLLOWING_TAGS = "There are no food with the following tags:";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Shows all food with the specified tag.\n"
+            + "Example: " + "\n"
+            + COMMAND_WORD + " n/Dairy" + "\n"
+            + COMMAND_WORD + " n/Dairy n/Healthy";
+
+    private final Set<Tag> targetTags;
+
+    /**
+     * Constructor to create a new ListByTag object.
+     */
+    public ListByTagCommand(Tag targetTag) {
```

Comment:
Perhaps can add a `requireNonNull` for the constructor parameter here? Same for the second constructor as well